open-authoring/OpenAuthoringErrorInterpreter.ts

```diff
@@ -0,0 +1,122 @@
+import OpenAuthoringError from "./OpenAuthoringError";
+import OpenAuthoringContextualError from "./OpenAuthoringContextualError";
+import { Actions } from "./OpenAuthoringModalContainer";
+import { isForkValid } from "./github/api";
+
+export default async function interpretError(error: OpenAuthoringError) : Promise<OpenAuthoringContextualError> {
```

Comment:
I think it feels a bit weird to me that this function is designed to return an error instance. Maybe instead of an error object, it's an "OpenAuthoringErrorUI" object that doesn't extend Error? 
If we want to treat OpenAuthoringContextualError as an actual error, I would expect the behaviour to catch a generic "OpenAuthoringError", and throw a more specific OpenAuthoringContextualError